src/aind_slims_api/operations/ecephys_session.py

```diff
@@ -0,0 +1,147 @@
+    """
+
+    session_group: SlimsExperimentRunStep
+    session_result: Optional[SlimsMouseSessionResult]
+    streams: Optional[List[SlimsStreamsResult]] = []
+    stream_modules: Optional[List[SlimsDomeModuleRdrc]] = []
+    reward_delivery: Optional[SlimsRewardDeliveryRdrc] = None
+    reward_spouts: Optional[SlimsRewardSpoutsRdrc] = None
+    stimulus_epochs: Optional[List[SlimsStimulusEpochsResult]] = []
+
+
+def _process_session_steps(
+        client: SlimsClient,
+        group_run_step: SlimsGroupOfSessionsRunStep,
+        session_run_steps: List[SlimsMouseSessionRunStep],
+) -> List[EcephysSession]:
+    """
+    Iterates through each session step and encapsulates all its related data
+    and return a list of EcephysSession objects.
+    """
+    ecephys_sessions = []
+
+    for step in session_run_steps:
+        # retrieve session, streams, and epochs from Results table
+        session = client.fetch_model(
+            SlimsMouseSessionResult, experiment_run_step_pk=step.pk
+        )
+        streams = client.fetch_models(
+            SlimsStreamsResult, mouse_session_pk=session.pk
+        )
+        stimulus_epochs = client.fetch_models(
+            SlimsStimulusEpochsResult, mouse_session_pk=session.pk
+        )
+
+        # retrieve modules and reward info from ReferenceDataRecord table
+        stream_modules = [
+            client.fetch_model(SlimsDomeModuleRdrc, pk=stream_module_pk)
+            for stream in streams
+            if stream.stream_modules_pk
+            for stream_module_pk in stream.stream_modules_pk
+        ]
+
+        reward_delivery = (
+            client.fetch_model(
+                SlimsRewardDeliveryRdrc, pk=session.reward_delivery_pk
+            )
+            if session.reward_delivery_pk
+            else None
+        )
+
+        reward_spouts = (
+            client.fetch_model(
+                SlimsRewardSpoutsRdrc, pk=reward_delivery.reward_spouts_pk
+            )
+            if reward_delivery and reward_delivery.reward_spouts_pk
+            else None
+        )
+
+        # encapsulate all info for a single session
+        ecephys_session = EcephysSession(
+            session_group=group_run_step,
+            session_result=session,
+            streams=streams or None,
+            stream_modules=stream_modules or None,
+            reward_delivery=reward_delivery,
+            reward_spouts=reward_spouts,
+            stimulus_epochs=stimulus_epochs or None,
+        )
+        ecephys_sessions.append(ecephys_session)
+
+    return ecephys_sessions
+
+
+def fetch_sessions(client: SlimsClient, subject_id: str) -> List[EcephysSession]:
+    """
+    Process all run steps for a given mouse
+     and return a list of EcephysSession objects.
+     Example Usage
+    --------
+    >>> from aind_slims_api import SlimsClient
+    >>> client = SlimsClient()
+    >>> fetch_sessions(client=client, subject_id="000000")
```

Comment:
Adding an example is great. Following valid doctest syntax may provide added benefit for little cost (makes the example testable). `>>> sessions = fetch_sessions(client=client, subject_id="725804")` But isn't a super necessary change.